### ISSUE TITLE
zsh bash completion issue

### ISSUE THREAD

**Issue by linsun** [istio/istio.io#4279]
Following the zsh bash completion issue in our get started page:

1. error on `source < _istioctl` cmd:
```
$ cp _istioctl ~/.
(⎈ |minikube:default)
~/Downloads/istio-1.2.0-rc.0/tools ⌚ 10:38:22
$ source < _istioctl 
source: not enough arguments
(⎈ |minikube:default)
~/Downloads/istio-1.2.0-rc.0/tools ⌚ 10:38:28
$ source _istioctl 
```

2. I find out `istioctl g` doesn't auto complete to `istioctl get`.  In chatting with @ericvn, he indicated hidden or deprecated cmds don't get this feature:

Eric Van Norman [11:23 AM]
Looking at the _cobra_ completion tests: https://github.com/spf13/cobra/blob/master/bash_completions_test.go#L223 hidden and deprecated do not get completion.

**Comment by linsun** [comment]
@ericvn or @ibm-jason can one of you own this?

**Comment by ibm-jason** [comment]
@ericvn, since it's zsh, probably makes the most sense for you to own this one.

**Comment by ericvn** [comment]
I think we can get the first portion in @ibm-jason's current PR. I have a few things to add for ZSH and will pick up the second comment with that PR.

**Comment by ericvn** [comment]
We will have to think about things going forward in the fall as MacOS Catalina will default to _zsh_ as the default shell. I don't know that it'll change much. I might change the tab ordering at that time to move MacOS with bash to the right.

**Comment by ericvn** [comment]
Should be fixed by #4312. Not sure why PRs aren't closing issues (probably something with my usage).
